### ISSUE TITLE
Barcode rotate bug

### ISSUE THREAD

**Issue by agkyunromb** [rxing-core/rxing#32]
### my code
~~~rust
use rxing::common::BitMatrix;
use rxing::helpers::save_file;
use rxing::pdf417::encoder::Dimensions;
use rxing::pdf417::PDF417Writer;
use rxing::BarcodeFormat::PDF_417;
use rxing::EncodeHintType::{ERROR_CORRECTION, MARGIN, PDF417_DIMENSIONS};
use rxing::EncodeHintValue::{ErrorCorrection, Margin, Pdf417Dimensions};
use rxing::EncodingHintDictionary as Hint;
use rxing::Writer;

fn main() {
    let not_rotate = short_code("test", 154, 80);
    save_file("not_roasted.jpg", &not_rotate).unwrap();
    let rotated = short_code("test", 80, 154);
    save_file("roasted.jpg", &rotated).unwrap();
}

fn short_code(contents: &str, width: i32, height: i32) -> BitMatrix {
    let writer = PDF417Writer::default();
    let dimensions = Dimensions::new(5, 5, 20, 20);
    let mut hints = Hint::new();
    hints.insert(ERROR_CORRECTION, ErrorCorrection("3".to_string()));
    hints.insert(PDF417_DIMENSIONS, Pdf417Dimensions(dimensions));
    hints.insert(MARGIN, Margin(("5").to_string()));

    writer
        .encode_with_hints(contents, &PDF_417, width, height, &hints)
        .unwrap()
}

~~~
#### not_rotated.jpg
![not_rotated](https://github.com/rxing-core/rxing/assets/26702058/5fdacfdc-32f5-4702-831c-ea8488ebb58c)
#### rotated.jpg
![rotated](https://github.com/rxing-core/rxing/assets/26702058/e71f3bcc-656c-4985-8ab5-d6cfc43fc771)

#### rotated.jpg
![roasted](https://github.com/rxing-core/rxing/assets/26702058/981a94f7-303c-496e-a235-2047d04c8a28)
#34

**Comment by hschimke** [comment]
Fixed by #34

**Comment by hschimke** [comment]
This fix appears in v0.4.6, currently uploading to cargo
